gorp_test.go

```diff
@@ -2360,6 +2361,31 @@ func TestPrepare(t *testing.T) {
 	}
 }
 
+func TestWithCanceledContext(t *testing.T) {
```

Comment:
> (2) The Context API being added to db.Sql wasn't added until I think Go 1.8. [...]  but that will make the test I added fail unless I can exclude that test for earlier versions.

If you'd put that into its own file, you would be able to add `// +build !go1.7`, I believe? To make these tests not run on 1.7? (not entirely sure if it works like that, though)